src/hdf5/HDF5IO.cpp

```diff
@@ -377,6 +379,14 @@ Status HDF5IO::createStringDataSet(const std::string& path,
   return Status::Success;
 }
 
+Status HDF5IO::startRecording()
+{
+  if (H5Fstart_swmr_write(this->file->getId()) < 0) {
+    return Status::Failure;
+  }
```

Comment:
One question to consider is whether a user would want to be able to configure whether to enable `swmr` or not? I think at least for now, always enabling is a good idea. I think the main consequence of this is that we are enforcing a particular use pattern of:

1. setup the NWB file
2. record to the NWB file without being able to modify the structure of the file
3. pause/stop the recording process to make modifications to the file structure

I think this would in general be a best practice for data acquisition and I suspect should work for most cases. However, I think it is useful to discuss this pattern when we do outreach with the acquisition system developers to see if there may be a need to diverge from this pattern at some point. For now, I would make an issue (low priority) to explore the option to make "SWMR" mode optional by allowing the user to provide a flag to `startRecording`.

Reply:
That sounds good, linked a low priority issue on this.